src/main/java/smokefree/projection/InitiativeProjection.java

```diff
@@ -115,11 +154,82 @@ void on(CheckListUpdateEvent evt) {
     }
 
     public Playground playground(String id, String userId) {
-        return playgrounds.containsKey(id) ? playgrounds.get(id).getPlaygroundForUser(userId) : null;
+        Playground playground = playgrounds.containsKey(id) ? playgrounds.get(id).getPlaygroundForUser(userId) : null;
+        return playground;
     }
 
     public Progress progress() {
         return progress;
     }
 
+    /**
+     * It finds the playgrounds list which are nearby to the given GeoLocation.
+     * So it will useful to display in front-end, asking user for confirmation
+     * @param newPlaygroundLocation
+     * @param distance
+     * @return
+     */
+    public List<Playground> checkForNearByPlaygrounds(GeoLocation newPlaygroundLocation, long distance) {
+        GeodeticCalculator geoCalc = new GeodeticCalculator();
```

Comment:
my preference is to spell out variables, at least the last noun, so either geodeticCalculator or geoCalculator